Validation/EventGenerator/plugins/GenWeightValidation.cc

```diff
@@ -0,0 +1,187 @@
+                "Number of Events");
+  bookTemplates(aDqmHelper,
+                leadJetEtaTemp_,
+                "leadJetEta",
+                "leading Gen jet #eta",
+                rapidityNbin_,
+                -rapidityRange_,
+                rapidityRange_,
+                "#eta_{j}",
+                "Number of Events");
+
+  return;
+}
+
+void GenWeightValidation::bookTemplates(DQMHelper& aDqmHelper,
+                                        std::vector<MonitorElement*>& tmps,
+                                        const std::string& name,
+                                        const std::string& title,
+                                        int nbin,
+                                        float low,
+                                        float high,
+                                        const std::string& xtitle,
+                                        const std::string& ytitle) {
+  tmps.push_back(aDqmHelper.book1dHisto(name, title, nbin, low, high, xtitle, ytitle));
+  tmps.push_back(aDqmHelper.book1dHisto(name + "FSRup", title + " FSR up", nbin, low, high, xtitle, ytitle));
+  tmps.push_back(aDqmHelper.book1dHisto(name + "FSRdn", title + " FSR down", nbin, low, high, xtitle, ytitle));
+  tmps.push_back(aDqmHelper.book1dHisto(
+      name + "FSRup_ratio", "Ratio of " + title + " FSR up / Nominal", nbin, low, high, xtitle, ytitle));
+  tmps.at(3)->setEfficiencyFlag();
+  tmps.push_back(aDqmHelper.book1dHisto(
+      name + "FSRdn_ratio", "Ratio of " + title + " FSR down / Nominal", nbin, low, high, xtitle, ytitle));
+  tmps.at(4)->setEfficiencyFlag();
+  tmps.push_back(aDqmHelper.book1dHisto(name + "ISRup", title + " ISR up", nbin, low, high, xtitle, ytitle));
+  tmps.push_back(aDqmHelper.book1dHisto(name + "ISRdn", title + " ISR down", nbin, low, high, xtitle, ytitle));
+  tmps.push_back(aDqmHelper.book1dHisto(
+      name + "ISRup_ratio", "Ratio of " + title + " ISR up / Nominal", nbin, low, high, xtitle, ytitle));
+  tmps.at(7)->setEfficiencyFlag();
+  tmps.push_back(aDqmHelper.book1dHisto(
+      name + "ISRdn_ratio", "Ratio of " + title + " ISR down / Nominal", nbin, low, high, xtitle, ytitle));
+  tmps.at(8)->setEfficiencyFlag();
+}  // to get ratio plots correctly - need to modify PostProcessor_cff.py as well!
+
+void GenWeightValidation::dqmBeginRun(const edm::Run&, const edm::EventSetup&) {}
+
+void GenWeightValidation::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  weights_ = wmanager_.weightsCollection(iEvent);
+
+  if (weights_.at(idxGenEvtInfo_).empty())
+    return;  // no weights in GenEventInfo
+
+  weight_ = weights_.at(idxGenEvtInfo_)[0] / std::abs(weights_.at(idxGenEvtInfo_)[0]);
+  nEvt_->Fill(0.5, weight_);
+  nlogWgt_->Fill(std::log10(weights_.at(idxGenEvtInfo_).size()), weight_);
+
+  for (unsigned idx = 0; idx < weights_.at(idxGenEvtInfo_).size(); idx++)
+    wgtVal_->Fill(weights_.at(idxGenEvtInfo_)[idx] / weights_.at(idxGenEvtInfo_)[1], weight_);
```

Comment:
Shouldn't this be
```C++
    wgtVal_->Fill(weights_.at(idxGenEvtInfo_)[idx] / weights_.at(idxGenEvtInfo_)[0], weight_);
```
instead?

Reply:
@perrotta this was intended on purpose since the proper normalization of PS weight is done by dividing weights to the baseline weight, which is located at idx 1 ([Twiki](https://twiki.cern.ch/twiki/bin/viewauth/CMS/HowToPDF#Parton_shower_weights)). But indeed I can place a protector for the case `weights_.at(idxGenEvtInfo_).size()=1` between other protectors for [`size()=0`](https://github.com/SanghyunKo/cmssw/blob/eaec2485fa65c0f4638d69035a245a491b14d106/Validation/EventGenerator/plugins/GenWeightValidation.cc#L125-L126) case and [`size()<=idxMax_`](https://github.com/SanghyunKo/cmssw/blob/eaec2485fa65c0f4638d69035a245a491b14d106/Validation/EventGenerator/plugins/GenWeightValidation.cc#L135-L136) case

(filling `weights_.at(idxGenEvtInfo_)[0]/weights_.at(idxGenEvtInfo_)[0]` for `size()=1` case is redundant)

```c++
  if (weights_.at(idxGenEvtInfo_).size()<2)
    return;  // no baseline weight in GenEventInfo

  for (unsigned idx = 0; idx < weights_.at(idxGenEvtInfo_).size(); idx++)
    wgtVal_->Fill(weights_.at(idxGenEvtInfo_)[idx] / weights_.at(idxGenEvtInfo_)[1], weight_);
```